packages/@react-aria/combobox/docs/useComboBox.mdx

```diff
@@ -0,0 +1,1166 @@
+    popoverRef,
+    listBoxRef,
+    state,
+    shouldUseVirtualFocus,
+    ...otherProps
+  } = props;
+
+  // Get props for the listbox. Prevent focus moving to listbox via shouldUseVirtualFocus
+  let {listBoxProps} = useListBox({
+    autoFocus: state.focusStrategy,
+    disallowEmptySelection: true,
+    shouldUseVirtualFocus,
+    ...otherProps
+  }, state, listBoxRef);
+
+  // Handle events that should cause the popup to close,
+  // e.g. blur, clicking outside, or pressing the escape key.
+  let {overlayProps} = useOverlay({
+    onClose: () => state.close(),
+    shouldCloseOnBlur: true,
+    isOpen: state.isOpen,
+    isDismissable: true
+  }, popoverRef);
+
+  // Add hidden <DismissButton> components at the start and end of the list
+  // to allow screen reader users to dismiss the popup easily.
+  return (
+    <div {...overlayProps} ref={popoverRef}>
+      <DismissButton onDismiss={() => state.close()} />
+      <ul
+        {...mergeProps(listBoxProps, otherProps)}
+        ref={listBoxRef}
+        style={{
+          position: 'absolute',
+          width: '100%',
+          margin: '4px 0 0 0',
+          padding: 0,
+          listStyle: 'none',
+          border: '1px solid gray',
+          background: 'lightgray'
+        }}>
+        {[...state.collection].map(item => (
+          <Option
+            shouldUseVirtualFocus
+            key={item.key}
+            item={item}
+            state={state} />
+        ))}
+      </ul>
+      <DismissButton onDismiss={() => state.close()} />
+    </div>
+  )
+}
+
+function Option({item, state, shouldUseVirtualFocus}) {
+  let ref = React.useRef();
+  let isDisabled = state.disabledKeys.has(item.key);
+  let isSelected = state.selectionManager.isSelected(item.key);
+  // Track focus via focusedKey state instead of with focus event listeners
+  // since focus never leaves the text input in a ComboBox
+  let isFocused = state.selectionManager.focusedKey === item.key;
+
+  // Get props for the option element. Prevent options from recieving true focus via shouldUseVirtualFocus.
+  let {optionProps} = useOption({
+    key: item.key,
+    isDisabled,
+    isSelected,
+    shouldSelectOnPressUp: true,
+    shouldFocusOnHover: true,
+    shouldUseVirtualFocus
+  }, state, ref);
+
+  let backgroundColor;
+  let color;
+
+  if (isSelected) {
+    backgroundColor = 'blueviolet';
+    color = 'white';
+  } else if (isFocused) {
+    backgroundColor = 'gray';
+    color = 'black';
+  } else if (isDisabled) {
+    backgroundColor = 'transparent';
+    color = 'gray';
+  }
+
+  return (
+    <li
+      {...optionProps}
+      ref={ref}
+      style={{
+        background: backgroundColor,
+        color: color,
+        padding: '2px 5px',
+        outline: 'none',
+        cursor: 'pointer'
+      }}>
+      {item.rendered}
+    </li>
+  );
+}
+///- end collapse -///
+// Using the same ComboBox component code from the first example...
+
+<ComboBox label="Favorite Animal" menuTrigger="focus">
```

Comment:
Should we have all three cases of `menuTrigger` as examples?

Reply:
IMO I feel like the description before the example is sufficient, happy for others to weigh in